realkd/utils.py

```diff
@@ -0,0 +1,166 @@
+import numpy as np
+import pandas as pd
+
+
+def contains_non_numeric(column):
+    """
+
+    >>> contains_non_numeric(['nan', '2'])
+    False
+    >>> contains_non_numeric(['hello', 2])
+    True
+    """
+    for item in column:
+        try:
+            float(item)
+        except ValueError:
+            return True
+    return False
+
+
+def get_generic_column_headers(data):
+    """Returns x0, x1, ... x{n} for the number of columns in the data
+
+    :param ~numpy.array data: input data
+
+    :return: list[str] labels
+
+    >>> import numpy as np
+    >>> get_generic_column_headers(np.array([[1, 4],[2, 3],[24, 31]]))
+    ['x0', 'x1']
+    """
+    if len(data.shape) == 1:
+        # Data is one dimentional (i.e. there's only one row)
+        return [f"x{n}" for n in range(data.shape[0])]
+    return [f"x{n}" for n in range(data.shape[1])]
+
+def validate_data(data, labels=None):
+    return data if isinstance(data, RealkdArrayLike) else RealkdArrayLike(data, labels)
+
+class RealkdArrayLike:
+    @staticmethod
+    def get_labels(data, labels=None):
+        if labels is not None:
+            return labels
+        elif hasattr(data, "columns"):
+            return data.columns.to_list()
```

Comment:
I probably could/should have embraced the duck typing and just set labels to data.columns, since it's iterable, however I do use the list-only method `index`. Do you think I made the right call here?

![image](https://user-images.githubusercontent.com/32168886/202848192-ab8c2afb-fce4-4aae-8c6c-9dc3c4c9f3ae.png)